main.go

```diff
@@ -25,6 +25,13 @@ func Run(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
 		return
 	}
 
+	// Get the source code from the request
+	zigSource, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, "reading body", http.StatusInternalServerError)
+		return
+	}
+
```

Comment:
You'll want to `defer r.Body.Close()` after this conditional to ensure that the body gets closed after it responds.